go/ethadapter/obscuro_rpc_client.go

```diff
@@ -62,6 +67,9 @@ func (c *obscuroWalletRPCClient) TransactionReceipt(hash gethcommon.Hash) (*type
 	var r types.Receipt
 	err := c.client.Call(&r, rpcclientlib.RPCGetTxReceipt, hash)
 	if err != nil {
+		if err == rpcclientlib.ErrNilResponse {
+			return nil, ErrReceiptNotFound
```

Comment:
Not 100% necessary to but feel it will be cleaner to have a specific error from this client, can be checked for by caller without depending on the upstream